clients/go/zbc/credentialsProvider_test.go

```diff
@@ -24,26 +25,50 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/status"
+	"io/ioutil"
 	"net"
+	"net/http"
+	"os"
 	"strings"
 	"testing"
 )
 
-func TestZeebeClientCredentialsProvider(t *testing.T) {
+// request data
+const clientId = "someClient"
+const clientSecret = "someSecret"
+const audience = "localhost"
+
+// response data
+const accessToken = "someToken"
+const tokenType = "Bearer"
+
+func TestOAuthCredentialsProvider(t *testing.T) {
 	// given
 	interceptor := &recordingInterceptor{}
-	lis, grpcServer := createServerWithInterceptor(interceptor.unaryClientInterceptor)
+	gatewayLis, grpcServer := createServerWithInterceptor(interceptor.unaryClientInterceptor)
 
-	go grpcServer.Serve(lis)
+	go grpcServer.Serve(gatewayLis)
 	defer func() {
 		grpcServer.Stop()
-		_ = lis.Close()
+		_ = gatewayLis.Close()
 	}()
 
-	credsProvider, err := NewZeebeClientCredentialsProvider("../resources/creds")
-	require.NoError(t, err)
+	authServerLis, err := createMockAuthorizationServer(t)
```

Comment:
Since this is a test, and we're always writing it...what about having the panic in that method (`createMockAuthorizationServer`)? The defer should remain here, makes total sense, but we could have that method just not return an error, i.e. they either return the object or we die.